File: frontend/src/types/api/product.ts
import type { Category } from '@/types/store/HomeType';

export interface Product {
  productId: string;
  name: string;
  price: number;
  description?: string;
  category?: string;
  imageUrl: string;
  sales?: number;
  createTime?: string;
  stock: number;
  status: number;
  isActive?: number;
  rating?: number;
  reviewCount?: number;
  tags: string;
  categoryId: string;
  brand: string;
}

export interface ProductResponse {
  data: Product[];
  list: Product[];
  total: number;
  pageNum: number;
  pageSize: number;
  pages: number;
  categoryId?: string;
}

export interface PaginationParams {
  page: number;
  size: number;
  category?: string | null;
  minPrice?: number;
  maxPrice?: number;
  hasStock?: boolean;
  hasDiscount?: boolean;
}

export interface PaginationResult<T> {
  list?: T[];
  data?: T[];
  items?: T[];
  total: number;
  currentPage: number;
}

export interface CategoryResponse {
  categories: Category[];
}

export interface ProductDetail extends Product {
  stock: number;
  brand?: string;
  categoryId?: string;
  description?: string;
  images?: string[];
}

File: backend/src/main/java/com/wyc21/entity/Product.java
package com.wyc21.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class Product extends BaseEntity {
    private String productId; // 改为String类型
    private String name;
    private String description;
    private BigDecimal price;
    private Integer stock;
    private String brand;
    private String tags;
    private String imageUrl;
    private BigDecimal rating;
    private Integer reviewCount;
    private Integer isActive;
   
   
    private String categoryId; // 改为String类型

    private LocalDateTime modifiedTime;

    public void setIsActive(Integer isActive) {
        this.isActive = isActive;
    }

    
}